news/refactor-lexer-parser.rst

```diff
@@ -0,0 +1,12 @@
+**Added:**
+
+**Changed:**
+* Privatise certain attributes of lexer/parser to minimise API surface
+
+**Deprecated:**
+
+**Removed:**
+
+**Fixed:**
+
+**Security:**
```

Comment:
this could be 

```
**Added:**

* <news item>

**Changed:**

* Privatise certain attributes of lexer/parser to minimise API surface

**Deprecated:**

* <news item>

**Removed:**

* <news item>

**Fixed:**

* <news item>

**Security:**

* <news item>
```